.github/workflows/build-test-javascript.jsonnet

```diff
@@ -134,17 +134,32 @@ local test_job = {
     {
       name: 'Test JS artifacts',
       run: |||
-        make -C js -j $(nproc) test
-        make -C js/tests
+          make -C js -j $(nproc) test
       |||
     },
+    {
+      name: 'Setup APT for xvfb',
```

Comment:
All these deps are for xvfb, which is essentially a virtual (x11) display, so VSCode tests can run, since they require a display. Unlike emacs vscode is lame and does not have a daemon mode

Reply:
but the deps are specifcally for testing in a headless ubuntu CI, not anywhere else, so I think I'd prefer to leave them in this workflow since that's the only time they matter, no need to abstract

Reply:
or sorry, the other deps are for some testing framework that are used under the hood, let me think about it